### ISSUE TITLE
[processor/k8sattributes] Use the already available internal docker.ParseImageName function

### ISSUE THREAD

**Issue by TylerHelmuth** [open-telemetry/opentelemetry-collector-contrib#36418]
Should we use the already available internal [docker.ParseImageName](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/common/docker/images.go#L23) function instead?

I would prefer to all components use the same strategy for common functionalities. Note that the shared function cannot parse images with digest at the moment and should be fixed beforehand: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36279

_Originally posted by @rogercoll in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36145#discussion_r1846455737_

**Comment by ChrsMark** [comment]
We could also consider using the https://pkg.go.dev/github.com/distribution/reference#Reference library as it was mentioned at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36145#discussion_r1847328541 by @spiffyy99.

**Comment by vordimous** [comment]
@TylerHelmuth  I would like to take on this issue.

**Comment by vordimous** [comment]
I believe my earlier changes are still relevant, but the PR was closed, and I am unable to reopen it: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/37152

**Comment by rogercoll** [comment]
> I believe my earlier changes are still relevant, but the PR was closed, and I am unable to reopen it: [#37152](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/37152)

Good catch, there are still a couple of receivers that need to be updated (aws/k8scluster).
